src/resmom/linux/mom_mach.c

```diff
@@ -8059,7 +8059,7 @@ pidcache_reset(pidcachetype_t *pidcache)
 		fclose(fd);
 	}
 
-	if (ncantstat)
+	if (!nprocs && ncantstat)
```

Comment:
I would like to understand what would happen if there were no entries in the cpusetdir. In that case, should we be returning -1?